vertx-web/src/test/java/io/vertx/ext/web/handler/StaticHandlerTest.java

```diff
@@ -252,6 +252,89 @@ public void testHttp2Push() throws Exception {
     latch.await();
   }
 
+  @Test
+  public void testSkipCompressionForMediaTypes() throws Exception {
+    Set<String> compressedMediaTypes = new HashSet<>();
+    compressedMediaTypes.add("image/jpeg");
+    stat.skipCompressionForMediaTypes(compressedMediaTypes);
+    router.route().handler(stat);
+
+    waitFor(4);
+    HttpServer httpServer = vertx.createHttpServer(new HttpServerOptions().setCompressionSupported(true));
+    httpServer.requestHandler(router).listen(8444);
+    HttpClient client = vertx.createHttpClient();
+
+    final List<CharSequence> acceptEncodings = new ArrayList<CharSequence>();
+    acceptEncodings.add("gzip");
+    acceptEncodings.add("jpg");
+    acceptEncodings.add("jpeg");
+    acceptEncodings.add("png");
+    final String[] routes = {"/testCompressionSuffix.html", "/somedir/range.jpg", "/somedir/range.jpeg", "/somedir3/coin.png"};
+    final int[] statusCodes = new int[4];
+    final String[] contentEncodings = new String[4];
+
+    for (int i = 0; i < 4; i++) {
+      int finalI = i;
+      HttpClientRequest request = client.get(8444, "localhost", routes[i], resp -> {
+        statusCodes[finalI] = resp.statusCode();
+        contentEncodings[finalI] = resp.getHeader(HttpHeaders.CONTENT_ENCODING);
+        complete();
+      });
+      request.putHeader(HttpHeaders.ACCEPT_ENCODING, acceptEncodings);
+      request.end();
+    }
+    await();
+
+    final int successCode = 200;
+    //jpeg and jpg have the same mime type, hence both are not compressed
+    final CharSequence[] expectedContentEncodings = {"gzip", HttpHeaders.IDENTITY, HttpHeaders.IDENTITY, "gzip"};
+    for (int i = 0; i < 4; i++) {
+      assertEquals("Status code does not equal 200 for request " + (i+1), successCode, statusCodes[i]);
+      assertEquals("Content-Encoding did not match for request " + (i+1), expectedContentEncodings[i].toString(), contentEncodings[i]);
+    }
+  }
+
+  @Test
+  public void testSkipCompressionForSuffixes() throws Exception {
+    Set<String> compressedSuffixes = new HashSet<>();
+    compressedSuffixes.add("jpg");
+    stat.skipCompressionForSuffixes(compressedSuffixes);
+    router.route().handler(stat);
+
+    waitFor(4);
+    HttpServer httpServer = vertx.createHttpServer(new HttpServerOptions().setCompressionSupported(true));
```

Comment:
Same comment applies here, close the existing `server` and assign the new one to it.